tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -15541,46 +15541,47 @@ void DiagnoseNodeEntry(Sema &S, FunctionDecl *FD, llvm::StringRef StageName,
           << NodeDG->getSpelling();
     }
-    // Node with NodeMaxDispatchGrid must have SV_DispatchGrid semantic.
-    if (NodeMDG) {
-      bool Found = false;
-      for (FunctionDecl::param_iterator I = FD->param_begin(),
-                                        E = FD->param_end();
-           I != E; ++I) {
-        QualType ParamType = (*I)->getType().getCanonicalType();
-
-        // Find parameter that is the node input record
-        if (hlsl::IsHLSLNodeInputType(ParamType)) {
-          // Node input records are template types
-          if (const RecordType *NodeInputRT = dyn_cast<RecordType>(ParamType)) {
-            if (const ClassTemplateSpecializationDecl *templateDecl =
-                    dyn_cast<ClassTemplateSpecializationDecl>(
-                        NodeInputRT->getDecl())) {
-
-              // Get the input record struct
-              auto &TemplateArgs = templateDecl->getTemplateArgs();
-              DXASSERT_NOMSG(TemplateArgs.size() >= 1);
-              QualType Arg0Type = TemplateArgs.get(0).getAsType();
-              const RecordType *NodeInputStructType =
-                  Arg0Type->getAsStructureType();
-              if (nullptr != NodeInputStructType) {
-                CXXRecordDecl *NodeInputStructDecl =
-                    dyn_cast<CXXRecordDecl>(NodeInputStructType->getDecl());
-                if (nullptr != NodeInputStructDecl) {
-                  // Make sure there is exactly one SV_DispatchGrid semantics
-                  // and it has correct type.
-                  DiagnoseDispatchGridSemantics(S, NodeInputStructDecl, Found);
-                }
+    // Diagnose dispatch grid semantics.
+    bool Found = false;
+    for (FunctionDecl::param_iterator I = FD->param_begin(),
+                                      E = FD->param_end();
+         I != E; ++I) {
+      QualType ParamType = (*I)->getType().getCanonicalType();
+
+      // Find parameter that is the node input record
+      if (hlsl::IsHLSLNodeType(ParamType)) {
+        // Node records are template types
+        if (const RecordType *NodeRT = dyn_cast<RecordType>(ParamType)) {
+          if (const ClassTemplateSpecializationDecl *templateDecl =
+                  dyn_cast<ClassTemplateSpecializationDecl>(
+                      NodeRT->getDecl())) {
+
+            // Get the record struct
+            auto &TemplateArgs = templateDecl->getTemplateArgs();
+            DXASSERT_NOMSG(TemplateArgs.size() >= 1);
+            QualType Arg0Type = TemplateArgs.get(0).getAsType();
+            if (const RecordType *NodeStructType =
+                    Arg0Type->getAsStructureType()) {
+              if (CXXRecordDecl *NodeStructDecl =
+                      dyn_cast<CXXRecordDecl>(NodeStructType->getDecl())) {
+                bool OutputFound = false;
+                // Make sure there is exactly one SV_DispatchGrid semantics
+                // and it has correct type.
```

Comment:
```suggestion
                // Diagnose any SV_DispatchGrid semantics used in record.
                // OutputFound provides the output needed by
                // DiagnoseDispatchGridSemantics, but isn't used otherwise.
                bool OutputFound = false;
```

This comment isn't quite right.  This code doesn't make sure there's exactly one, it makes sure there is at most one and that it's on a field of a supported type.  The later code makes sure one was found on the input when it's required.

I've suggested a change, and moved `OutputFound`, making it clear why it's needed, and why it isn't used elsewhere.

Reply:
Strangely, GitHub won't show this comment in file view.  Is it because it includes or ends in lines that aren't changed in the commit?

Reply:
Fixed.